cas-server-core/src/main/java/org/jasig/cas/AbstractCentralAuthenticationService.java

```diff
@@ -150,6 +159,15 @@ protected final void doPublishEvent(final ApplicationEvent e) {
 
 
     /**
+     * Gets a synchronization handle.
+     * @param id String mutex id
+     * @return SimpleMutex
+     */
+    protected final SimpleMutex getMutex(final String id) {
```

Comment:
Also, any specific reason you went with creating a `SimpleMutex` class instead of just using a simple `new Object()` for the mutex?